src/utiltime.cpp

```diff
@@ -9,39 +9,93 @@
 #endif
 
 #include "utiltime.h"
+#include "sync.h"
 
 #include <chrono>
 #include <boost/date_time/posix_time/posix_time.hpp>
 #include <boost/thread.hpp>
 
 using namespace std;
 
-static int64_t nMockTime = 0; //!< For unit testing
+RecursiveMutex clock_lock;
```

Comment:
Awww, I liked 'clock_lock'. :)

```suggestion
RecursiveMutex cs_clock;
```